MabiPale2.Plugins.PacketAnalyzer/Packets/ConditionUpdate.cs

```diff
@@ -68,7 +65,15 @@ public string AnalyzePacket(PalePacket palePacket)
 				var active = palePacket.Packet.GetBool();
 				var conditionId = palePacket.Packet.GetInt();
 
-				sb.AppendLine("Condition Id: {0} ({1})", (ConditionId)conditionId, conditionId);
```

Comment:
This seems like a roundabout way of getting the name of the condition, and testing this code I'm getting an exception. What was wrong with the cast, which results in the name as well?

Reply:
ConditionId isn't an enum in the new Mabi.dll as it was in the old one, so it's not something that can be casted to. Every effect id is a static readonly int belonging to the class ConditionId. To get the name, I just got the names of those variables instead. Where exactly are you getting an exception though? I did the try/catch block to avoid that situation...

Reply:
The previous version of Pale had a version of Mabi.dll that had the enums for ConditionsA, ConditionsB, and etc. It also had its own ConditionId.cs file containing a ConditionId enum that was independent of Mabi.dll, since Mabi.dll did not support the new condition system at the time.

The new Mabi.dll does **not** have the ConditionsA, B, etc enums, but it has its own definition of ConditionId instead, to support the new condition system.

Basically, what I did was delete Pale's ConditionId enum as it was redundant. Mabi.dll has its own definition for that now. Mabi.dll lacks the old conditions though, so I added those into Pale to make up for what was lost.

The problem that arose, though, was that in Mabi.dll, ConditionId is a **static class**, not an enum like it was in the Pale ConditionId file. I can't cast to it. I decided to get the names of the fields instead, however hacky that may be.

I don't know why you're getting an exception though; I did the try/catch block to avoid that situation and had no errors on my end...

Reply:
Huh, why did we use const ints there o.o This explains why I was getting an exception though, because I didn't remember that^^ I wasn't at my PC and tested the snippet online with an enum. However, you will still get an exception if a condition doesn't exist, like when Mabi.dll wasn't updated for a new condition yet, which we should probably check for. Maybe something like this? It will return "?" if the condition doesn't exist in ConditionId.
```cs
var name = typeof(ConditionId).GetFields().FirstOrDefault(a => (int)a.GetValue(null) == conditionId)?.Name ?? "?";
```

Reply:
Sure, I'll add that in. I used a try/catch block to catch the exception and leave the name blank, but this is much cleaner so I'll update it right now :^)